## task1
Create a two-dimensional array of size 4×4, filled with random integers from 1 to 50.

- Display the array in a readable way (as a table).
- Find the sum of elements for each row and each column (use the axis parameter).
- Determine the overall average value of the array.
- Output all results in a clear and readable format (for example: “Row sums: …”, “Column sums: …”).

In [6]:
import numpy as np

arr = np.random.randint(1, 51, size=(4, 4))

print("Original array (4×4):")
print(arr)

row_sums = np.sum(arr, axis=1)

col_sums = np.sum(arr, axis=0)

mean_value = np.mean(arr)


print("\nRow sums:", row_sums)
print("Column sums:", col_sums)
print("Mean value of the array:", mean_value)


Original array (4×4):
[[35 42  7 13]
 [44  1 23 18]
 [20 26 50  8]
 [20 17 23 32]]

Row sums: [ 97  86 104  92]
Column sums: [119  86 103  71]
Mean value of the array: 23.6875


## task2
Create an array of size 3×5, filled with numbers from 0 to 14.
• Display the original array.
• Perform a transposition of the array (rows become columns and columns become rows).
• Apply the reshape operation to obtain a 5×3 array.
• Display all results on the screen.


In [7]:
import numpy as np


arr = np.arange(15).reshape(3, 5)

print("Original array (3×5):")
print(arr)

transposed = arr.T
print("\nTransposed array (5×3):")
print(transposed)

reshaped = arr.reshape(5, 3)
print("\nReshaped array (5×3):")
print(reshaped)


Original array (3×5):
[[ 0  1  2  3  4]
 [ 5  6  7  8  9]
 [10 11 12 13 14]]

Transposed array (5×3):
[[ 0  5 10]
 [ 1  6 11]
 [ 2  7 12]
 [ 3  8 13]
 [ 4  9 14]]

Reshaped array (5×3):
[[ 0  1  2]
 [ 3  4  5]
 [ 6  7  8]
 [ 9 10 11]
 [12 13 14]]


## task3

- Create a Markdown cell with the title "Sales Report" and add the subtitle:
  *"In this section, we calculate basic statistics for simple data."*

- In a code cell, create a list of weekly sales:
  `sales = [120, 135, 100, 150, 90, 200, 170]`

- Calculate:
  - the total sales for the week;
  - the average value;
  - the maximum and minimum values.

- Create a Markdown cell and format a table with the results:


In [8]:
sales = [120, 135, 100, 150, 90, 200, 170]

total_sales = sum(sales)
average_sales = sum(sales) / len(sales)
maximum_sales = max(sales)
minimum_sales = min(sales)

print("Total sales:", total_sales)
print("Average sales:", average_sales)
print("Maximum sale:", maximum_sales)
print("Minimum sale:", minimum_sales)


Total sales: 965
Average sales: 137.85714285714286
Maximum sale: 200
Minimum sale: 90


## Summary Table

| Metric        | Value         |
|---------------|---------------|
| Total Sales   | 965           |
| Average       | 137.86        |
| Maximum       | 200           |
| Minimum       | 90            |
